compiler/rustc_trait_selection/src/solve/fulfill/derive_errors.rs

```diff
@@ -291,6 +291,34 @@ impl<'tcx> BestObligation<'tcx> {
         }
     }
 
+    /// When a higher-ranked projection goal fails, check that the corresponding
+    /// higher-ranked trait goal holds or not. This is because the process of
+    /// instantiating and then re-canonicalizing the binder of the projection goal
+    /// forces us to be unable to see that the leak check failed in the nested
+    /// `NormalizesTo` goal, so we don't fall back to the rigid projection check
+    /// that should catch when a projection goal fails due to an unsatisfied trait
+    /// goal.
+    fn detect_error_in_higher_ranked_projection(
```

Comment:
```suggestion
    fn detect_trait_error_in_higher_ranked_projection(
```

r=me with this change

Reply:
Already got rolled up, and I don't think this is a significant enough change to kick it out of the rollup. I'll fold this into #139513 which is based off this one.

Reply:
#139564 *

Reply:
Unless this fails to merge in which case I'll amend this. I'll keep an eye on the rollup lol